### ISSUE TITLE
Cypress CT - fails on a custom TS webpack config

### ISSUE THREAD

**Issue by danr-za** [nrwl/nx#13178]
## Current Behavior

When running a component test for an app that has a TS webpack config (e.g `webpack.config.ts`) it fails on TS errors, which results in using the fallback config.

## Expected Behavior

Should be able to compile the config and run with it

## Steps to Reproduce

* Clone this repro - https://github.com/danr-za/webpack-ct-ts
* Run `nx run my-app:component-test`

Notice the console error:
```shell
Unable to build a webpack config with the project graph.
Falling back to default webpack config.
TSError: ⨯ Unable to compile TypeScript:
webpack.config.ts(3,20): error TS7006: Parameter 'configuration' implicitly has an 'any' type.
```

This fails when resolving the TS here: https://github.com/nrwl/nx/blob/54670c93e0e35b041d0b6a2d1a0057e5ec109782/packages/webpack/src/utils/webpack/custom-webpack.ts#L1-L19

This issue may not be prioritized if details are not provided to help us reproduce the issue.

**Comment by danr-za** [comment]
Hey folks, any news? 🙏

**Comment by danr-za** [comment]
@barbados-clemens did you have the change to check it out?

**Comment by barbados-clemens** [comment]
I'm not too sure as I'm not famailar with the inner workings of nrwl/webpack and why it might not be set that way. 

I know there are plans to clean up nrwl/webpack in the coming months. So let me sync with @jaysoo to see what needs to be done here. 

But the error also doesn't make sense to me or I'm missing something. 

```
webpack.config.ts(3,20): error TS7006: Parameter 'configuration' implicitly has an 'any' type.
```

isn't configuration typed as "Configuration" so why would it have an implicit type of 'any'?

I'll have to look more into this as maybe it's a setup thing that I missed for CT.

**Comment by danr-za** [comment]
The `configuration` is what I use in the exported function:
```ts
export default function (configuration: Configuration) { ... }
```

one of the issues is that it somehow resolves it as `any`, no idea why

**Comment by barbados-clemens** [comment]
Same here, 
work around would be to make it a .js file, turn off implicitAny or just use ts-ignore until I can figure out what is happening here. 

the _sort of_ bandaid solution would be to turn off the type checking for CT, but also would make sense that people would want to be on.

**Comment by danr-za** [comment]
unfortunately `.js` is not an option in our case, we currently use vite for CT, but that's a temporary patch.
I think that registering `ts-node` dynamically can be safely done without typechecking, and let the users handle it on their side.
Or, use the same configuration from `@nrwl/webpack:webpack` that does not throw any errors.

**Comment by barbados-clemens** [comment]
yeah I'd recommend just setting a // @ts-ignore above for now, until I can figure out more info as to what on earth is happening.

**Comment by barbados-clemens** [comment]
Hi @danr-za, sorry got getting back to this so late!

I just retried this with an updated workspace (migrated your example to Nx latest, which includes Cypress v12) and now the behavior is gone. 

I believe this might have been because of ts-node being double registered, which was fixed a bit ago in @nrwl/webpack. 

Here is a PR of the changes against your original example. 
https://github.com/danr-za/webpack-ct-ts/pull/1
